crates/core/src/nats/conn_streams.rs

```diff
@@ -0,0 +1,70 @@
+use crate::nats::{
+    streams::Stream,
+    subjects::{BlockSubjects, TransactionSubjects},
+    NatsClient,
+    NatsError,
+};
+
+#[derive(Debug, Clone)]
+pub struct ConnStreams {
+    pub blocks: Stream<BlockSubjects>,
+    pub transactions: Stream<TransactionSubjects>,
+}
+
+impl ConnStreams {
+    pub async fn new(client: &NatsClient) -> Result<Self, NatsError> {
+        let transactions = Stream::<TransactionSubjects>::new(client).await?;
+        let blocks = Stream::<BlockSubjects>::new(client).await?;
+
+        Ok(Self {
+            transactions,
+            blocks,
+        })
+    }
+}
+
+#[cfg(any(test, feature = "test_helpers"))]
+impl ConnStreams {
+    pub fn get_stream_list(&self) -> Vec<super::types::AsyncNatsStream> {
+        vec![self.blocks.stream.clone(), self.transactions.stream.clone()]
+    }
+
+    pub async fn collect_subjects(
+        streams: Vec<super::types::AsyncNatsStream>,
+    ) -> anyhow::Result<Vec<String>> {
+        let mut all_subjects = Vec::new();
+        for mut stream in streams {
+            let info = stream.info().await?;
+            all_subjects.extend(info.config.subjects.clone());
+        }
+
+        Ok(all_subjects)
+    }
```

Comment:
Nit: Not sure why this is embedded in `ConnStream`'s impl (?) Since it's a test helper that takes an AsynNatsStream type

Reply:
It's been used outside of this file within the `TestStreamBuilder`, but I'll change it indeed to simplify its usage